### PR TITLE
reject fragments that will overflow the fragments pool

### LINKED ISSUE #3064: Change from oldest fragment eviction to rejecting new fragments when the pool is full
When the leadership task's throughput for processing incoming fragments is exceeded and the pool has been filled to the limit, some fragments need to be rejected. The current behavior of evicting the least recently added fragments is problematic:
* Some newer fragments may optimistically depend on fragments added to the pool earlier.
* A dishonest participant can attempt to flush out an unwanted transaction after observing it in the pool. It does not cost them anything to fill up the pool with invalid transactions.

Instead, the leadership task should switch to rejecting any new incoming transactions while the pool capacity is full.

### PR DESCRIPTION
**Pull request by eugene-babichenko**
closes #3064

### CODE REVIEW

**Review by mzabaluev — COMMENTED** [review]
Looks like `Pool::insert` still does not enforce the capacity limit. Maybe remove that method if it's not used anywhere.

File: jormungandr/Cargo.toml [hunk 1]
```diff
@@ -73,6 +73,11 @@ features = ["rustls-tls"]
 [dev-dependencies]
 rand_core = "0.5"
 tokio = { version = "^1.0", features = ["full"] }
+quickcheck = "0.9"
+quickcheck_macros = "0.9"
```
Comment (mzabaluev) [inline jormungandr/Cargo.toml:77]: It's better to use quickcheck 1.0 while we are at it, possibly integrating #2927, https://github.com/input-output-hk/chain-libs/pull/506, and https://github.com/input-output-hk/chain-libs/pull/507. Unless this is too much work and requires another PR.

Comment (eugene-babichenko) [reply jormungandr/Cargo.toml:76]: I think it does require another PR, there are breaking interface changes.

**Review by mzabaluev — APPROVED** [review]
LGTM

With this in, does `self.entries` still need to be an LRU?

**Comment by eugene-babichenko** [comment]
Can be turned into a simple queue actually.

**Pull request merged**